Sources/SmithyReadWrite/WireResponseClosure.swift

```diff
@@ -0,0 +1,29 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+public typealias WireResponseOutputClosure<WireResponse, OperationStackOutput> =
+    (WireResponse) async throws -> OperationStackOutput
+
+public typealias WireResponseOutputBinding<
+    WireResponse: WireDataProviding,
+    OperationStackOutput,
+    Reader: SmithyReader> =
+        (WireResponse, WireResponseDocumentBinding<WireResponse, Reader>) async throws -> OperationStackOutput
```

Comment:
These typealiases replace the former `HTTPResponseOutputClosure` and `HTTPResponseOutputBinding` with closures that are not dependent on HTTP.

Otherwise, this is just moved in from ClientRuntime.